File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/service/ProductService.java
package com.firefly.example.reactive.coffee.store.service;

import com.firefly.example.reactive.coffee.store.model.Product;
import com.firefly.example.reactive.coffee.store.vo.Page;
import com.firefly.example.reactive.coffee.store.vo.ProductQuery;
import reactor.core.publisher.Mono;

/**
 * @author Pengtao Qiu
 */
public interface ProductService {

    Mono<Page<Product>> list(ProductQuery query);

}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/service/ProductService.kt
package com.firefly.example.kotlin.coffee.store.service

import com.firefly.example.kotlin.coffee.store.model.Product
import com.firefly.example.kotlin.coffee.store.vo.Page
import com.firefly.example.kotlin.coffee.store.vo.ProductQuery

/**
 * @author Pengtao Qiu
 */
interface ProductService {
    suspend fun list(query: ProductQuery): Page<Product>
}